### ISSUE TITLE
Memory leak when clearing and redrawing 3D scene group

### ISSUE THREAD

**Issue by h0uter** [zauberzeug/nicegui#727]
### Description

Hey,

I'm really enjoying nicegui, however, I ran into an issue with rendering 3D scenes.

1. I want to clear and rerender a 3D scene from scratch with a specific frequency. (robotics exploration/navigation graph)
However, my browser's memory consumption keeps increasing until it crashes, indicating a memory leak somewhere.

<img width="500" alt="image" src="https://user-images.githubusercontent.com/40892865/230148935-8de590fa-8818-434a-8c05-c5c1bb44fe5a.png">

I use a scene_group because it allows deleting all contained scene objects in one go.
however `len(self.my_scene.objects)` keeps increasing after calling `delete()` on the scene group.

<img width="398" alt="image" src="https://user-images.githubusercontent.com/40892865/230149072-d961e488-330d-4958-a64c-487d334453c8.png">

The below colorful code example reproduces the memory leak and this scene.
<img width="300" alt="image" src="https://user-images.githubusercontent.com/40892865/230151105-e00c98d2-edfe-4c96-b670-bf0cd9d5514a.png">

```python
import sys

import numpy as np
from nicegui import ui

NUMBER_OF_NEW_NODES = 400
RENDER_INTERVAL_S = 0.5


class MemoryLeakExample:
    """This example renders nodes in a scene, then clears the scene and renders new nodes."""

    def __init__(self) -> None:
        self.scene_node_objects = []
        self.color_toggle = "green"
        self.my_scene_group = None

        self.make_page_layout()
        ui.run(title="Memory Leak example")

    def make_page_layout(self):
        with ui.scene(width=1000, height=700) as self.my_scene:
            ui.timer(
                interval=RENDER_INTERVAL_S, callback=self.render_scene, active=True
            )

    def render_scene(self):
        print(f"number of scene.objects before clear: {len(self.my_scene.objects)}")
        print(
            f"getsizeof() scene.objects before clear: {sys.getsizeof(self.my_scene.objects)}"
        )

        self.clear_scene()  # this clears elements from the scene, but does not free up memory

        print(f"number of scene.objects after clear: {len(self.my_scene.objects)}")
        print(
            f"getsizeof() scene.objects after clear: {sys.getsizeof(self.my_scene.objects)}"
        )

        """Toggle the color of the new nodes for visual distinction"""
        if self.color_toggle == "green":
            self.color_toggle = "red"
        else:
            self.color_toggle = "green"

        """I create a scene group so I can clear all scene objects in one sweep."""
        with self.my_scene.group() as self.my_scene_group:

            random_node_postions = (
                (np.random.randint(-50, 50), np.random.randint(-50, 50))
                for _ in range(NUMBER_OF_NEW_NODES)
            )
            for pos in random_node_postions:
                self.add_node(x=pos[0], y=pos[1])

    def add_node(self, x, y):
        """adds a node scene object to the scene"""
        scene_object = self.my_scene.sphere(radius=0.5)
        scene_object.move(x=x, y=y)
        scene_object.material(self.color_toggle)

    def clear_scene(self):
        if self.my_scene_group:
            self.my_scene_group.delete()  # this does not seem to free up memory
            self.my_scene_group = None


if __name__ in {"__main__", "__mp_main__"}:
    MemoryLeakExample()
```

I already tried manually tracking and deleting all the scene objects in the scene group, but while that stops `len(self.my_scene.objects)` from growing, the memory leak remains.

I wanted to check what the recommended way was to use nicegui for clearing and rendering 3D scenes like this?

---
2. I expected the memory to be freed when calling `self.my_scene_group.delete()`
---
3. Instead the memory consumption of my browser keeps increasing until it crashes.

**Comment by falkoschindler** [comment]
Wow, what a severe bug! Thanks for reporting, @h0uter!
We're using 3D scenes in several of our robotics projects, but usually replace individual objects, not groups as a whole. I guess a workaround would be to delete all spheres contained in a group before deleting the group itself. But of course, we should fix this bug in NiceGUI, since deleting a group without its children doesn't make too much sense.

Here is a more compact reproduction based on your code. Eventually we should bake it into a pytest.

```py
def replace_group():
    global group
    print(f'before clear: {len(scene.objects) = }')
    if group:
        group.delete()
    print(f'after clear:  {len(scene.objects) = }')
    with scene.group() as group:
        scene.sphere()

group: ui.scene.group = None
with ui.scene() as scene:
    ui.timer(1.0, replace_group)
```

**Comment by falkoschindler** [comment]
Ok, should be fixed with commit [d53ac01](https://github.com/zauberzeug/nicegui/commit/d53ac0116de6847edbf51caec2d60a939ddfb083).
The pytest from [c1f0c16](https://github.com/zauberzeug/nicegui/commit/c1f0c164cbab6f196bc0551a7949d98f5e45c97e) currently only runs with `--driver firefox` in pytest.ini, until we fix #643.

**Comment by h0uter** [comment]
Cheers, awesome response time! 
Will use the fix until the release of the next version.
